### PR TITLE
Pause protocol flow

### PR DESCRIPTION
**Pull request by vito-kovalione**
Introduce AdminControls contract responsible for pausing and unpausing protocol.

### CODE REVIEW

File: contracts/main/stablecoin-core/AdminControls.sol [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        systemDebtEngine = _systemDebtEngine;
+        stablecoinAdapter = _stablecoinAdapter;
+    }
+
+    function pauseProtocol() external onlyOwnerOrGov {
+        IPausable(bookKeeper).pause();
+        IPausable(positionManager).pause();
+        IPausable(liquidationEngine).pause();
+        IPausable(systemDebtEngine).pause();
+        IPausable(stableSwapModule).pause();
```
Comment (codinghistorian) [inline contracts/main/stablecoin-core/AdminControls.sol:59]: I think StableSwapModule should keep going although the protocol is paused to ensure FXD pegging at all time.

Comment (vito-kovalione) [reply contracts/main/stablecoin-core/AdminControls.sol:59]: agree, will change

Comment (codinghistorian) [reply contracts/main/stablecoin-core/AdminControls.sol:59]: Thank you bossue (_ _ )

Comment (vito-kovalione) [reply contracts/main/stablecoin-core/AdminControls.sol]: fixed

**Pull request merged**